src/main/java/net/sf/marineapi/nmea/io/AbstractDataReader.java

```diff
@@ -93,6 +93,13 @@ public void run() {
 		while (isRunning) {
 			try {
 				String data = read();
+				if (data == null) {
```

Comment:
Actually, only DefaultDataReader.java can return null here. UDPDataReader currently can not detect connection failure so it will not return null.